File: tests/fixtures/openai-cua/openai-cua-no-domain-allowlist-fixed/origin-comparison-before-fill.ts
// Looks like the broken shape at a glance (no variable literally named
// "allowlist"), but it does compare the live page's origin against the
// configured project origin before filling anything.
export async function fillSensitiveField(page: any, projectOrigin: string, selector: string, value: string) {
  const currentUrl = new URL(page.url());
  if (currentUrl.origin !== projectOrigin) {
    throw new Error('Refusing to fill a field on an unexpected origin');
  }
  await page.fill(selector, value);
}

File: tests/fixtures/openai-cua/openai-cua-no-domain-allowlist-fixed/origin_comparison_before_fill.py
class ComputerAgent:
    ALLOWED_DOMAIN_LIST = ("example.com",)

    def run_step(self, action):
        if action.get("type") == "fill":
            if self.page.url_origin not in self.ALLOWED_DOMAIN_LIST:
                return
            self.page.fill(action["selector"], action["value"])
